File: problem015/15.rb
#!/usr/bin/env ruby

def factorial(n)
  1.upto(n).reduce(1, :*)
end

#http://mathworld.wolfram.com/CentralBinomialCoefficient.html
#we want to find the number of routes with the same number 
#of downs and rights, or ones and zeros
def routes(n)
  factorial(2*n) / (factorial(n) ** 2)
end

puts routes(20)

File: problem015/15.js
#!/usr/bin/env node

function factorial(n) {
    var i = 1;
    var f = 1;
    for (i; i <= n; i++) {
        f *= i;
    }
    return f;
}

function routes(n) {
    return factorial(2 * n) / (Math.pow(factorial(n), 2));
}

console.log(routes(20));
